code/game/machinery/fabricator.dm

```diff
@@ -0,0 +1,569 @@
+#define FABRICATOR_MAIN_MENU       1
+#define FABRICATOR_CATEGORY_MENU   2
+#define FABRICATOR_SEARCH_MENU     3
+#define FABRICATOR 5
+/obj/machinery/fabricator
```

Comment:
Hey its all good. For the future though, a child of the autolathe would of worked a bit better